tmt/beakerlib.py

```diff
@@ -217,7 +217,7 @@ def fetch(self) -> None:
             self.parent.debug(f"Fetch library '{self}'.", level=3)
             # Prepare path, clone the repository, checkout ref
             assert self.parent.workdir
-            directory = os.path.join(self.parent.workdir, self.dest, self.repo)
+            directory = str(Path(self.parent.workdir).joinpath(self.dest, self.repo))
```

Comment:
Still a draft, yet I'm going to add a comment or two anyway:

* `str(Path(DEFAULT_REPOSITORY).joinpath(self.repo))`: creating a `Path` from `DEFAULT_REPOSITORY` does not make much sense, as `DEFAULT_REPOSITORY` is indeed an URL and should be treated as such. For adding components to URLs, we have https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin: `self.url = urllib.parse.urljoin(DEFAULT_REPOSITORY, self.repo)` - using `os.path.join` for this was incorrect, and using `Path` would just preserve the mistake.
* `self.path` should become `Optional[Path]`
* `repo = Path(self.path).name` - `self.path` should already be a `Path` instance, that's the whole idea :) Therefore `repo = self.path.name` is probably better
* unless, of course, `self.repo` is actually a secret path! If it holds a path to a repository somewhere on the local filesystem, then `self.repo` itself should really turn into `Path` as well.
* `directory = str(Path(self.parent.workdir).joinpath(self.dest, self.repo))` - `self.parent.workdir` should turn into `Path`, too, as well as `self.dest` and `self.repo` (it was a secret path after all! :) and `directory` would become simpler, `directory = self.parent.workdir.joinpath(self.dest, self.repo)` - `directory` itself would be a `Path` instance, because it *is* a path, not just some generic, random, free-to-use `str`ing.